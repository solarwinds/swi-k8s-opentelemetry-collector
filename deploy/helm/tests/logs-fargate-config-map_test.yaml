# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for logs-fargate-config-map
templates:
  - logs-fargate-config-map.yaml
tests:
  - it: Fargate logging ConfigMap spec should match snapshot when Fargate logging is enabled
    template: logs-fargate-config-map.yaml
    set:
      aws_fargate.enabled: true
      aws_fargate.logs.enabled: true
      aws_fargate.logs.region: test-region
    asserts:
      - matchSnapshot:
          path: data
  - it: Fargate logging ConfigMap should not be generated when using default values
    template: logs-fargate-config-map.yaml
    asserts:
      - hasDocuments:
          count: 0
  - it: Fargate logging ConfigMap spec should include additional filters when they are configured in values.yaml
    template: logs-fargate-config-map.yaml
    set:
      aws_fargate.enabled: true
      aws_fargate.logs.enabled: true
      aws_fargate.logs.region: test-region
      aws_fargate.logs.filters: |
        [FILTER]
            Name filter_name
            Match *
    asserts:
      - matchSnapshot:
          path: data