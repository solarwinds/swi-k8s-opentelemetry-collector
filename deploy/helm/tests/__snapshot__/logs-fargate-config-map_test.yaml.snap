Fargate logging ConfigMap spec should include additional filters when they are configured in values.yaml:
  1: |
    filters.conf: |
      [FILTER]
          Name parser
          Match *
          Key_name log
          Parser crio
      [FILTER]
          Name kubernetes
          Match kube.*
          Merge_Log On
          Keep_Log Off
          Buffer_Size 0
          Kube_Meta_Cache_TTL 300s
          Labels Off
          Annotations Off
      [FILTER]
          Name filter_name
          Match *

      [FILTER]
          Name modify
          Match *
          Add sw.k8s.cluster.uid <CLUSTER_UID>
          Add sw.k8s.log.type container
          Add sw.k8s.agent.manifest.version "2.8.0-alpha.8"
    flb_log_cw: "false"
    output.conf: |
      [OUTPUT]
          Name cloudwatch_logs
          Match kube.*
          region test-region
          log_group_name /swo/fargate/cluster_name
          log_stream_prefix from-fluent-bit-
          log_retention_days 30
          auto_create_group true
    parsers.conf: |-
      [PARSER]
          Name crio
          Format Regex
          Regex ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>P|F) (?<log>.*)$
          Time_Key time
          Time_Format %Y-%m-%dT%H:%M:%S.%L%z
Fargate logging ConfigMap spec should match snapshot when Fargate logging is enabled:
  1: |
    filters.conf: |
      [FILTER]
          Name parser
          Match *
          Key_name log
          Parser crio
      [FILTER]
          Name kubernetes
          Match kube.*
          Merge_Log On
          Keep_Log Off
          Buffer_Size 0
          Kube_Meta_Cache_TTL 300s
          Labels Off
          Annotations Off
      [FILTER]
          Name modify
          Match *
          Add sw.k8s.cluster.uid <CLUSTER_UID>
          Add sw.k8s.log.type container
          Add sw.k8s.agent.manifest.version "2.8.0-alpha.8"
    flb_log_cw: "false"
    output.conf: |
      [OUTPUT]
          Name cloudwatch_logs
          Match kube.*
          region test-region
          log_group_name /swo/fargate/cluster_name
          log_stream_prefix from-fluent-bit-
          log_retention_days 30
          auto_create_group true
    parsers.conf: |-
      [PARSER]
          Name crio
          Format Regex
          Regex ^(?<time>[^ ]+) (?<stream>stdout|stderr) (?<logtag>P|F) (?<log>.*)$
          Time_Key time
          Time_Format %Y-%m-%dT%H:%M:%S.%L%z
